rust/worker/src/execution/operators/merge_metadata_results.rs

```diff
@@ -164,16 +164,8 @@ impl Operator<MergeMetadataResultsOperatorInput, MergeMetadataResultsOperatorOut
         };
 
         // Step 1: Materialize the logs.
-        let mut offset_id = Arc::new(AtomicU32::new(1));
-        match record_segment_reader.as_ref() {
-            Some(reader) => {
-                offset_id = reader.get_current_max_offset_id();
-                offset_id.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
-            }
-            None => {}
-        };
         let materializer =
-            LogMaterializer::new(record_segment_reader, input.filtered_log.clone(), offset_id);
+            LogMaterializer::new(record_segment_reader, input.filtered_log.clone(), None);
```

Comment:
Why were we loading it before?

Reply:
We still need to read the curr_max_offset_id from the record segment. For readers, the onus of doing that has shifted to the materializer now. And writers need to still pass it explicitly since all the writer threads need to share the same curr_max_offset_id.